pytest_factoryboy/fixture.py

```diff
@@ -96,8 +98,22 @@ def register_(factory_class: F) -> F:
         args = []
         attr_name = SEPARATOR.join((model_name, attr))
 
-        if isinstance(value, factory.declarations.PostGeneration):
-            value = kwargs.get(attr, None)
+        if isinstance(value, factory.declarations.PostGenerationDeclaration) and not isinstance(
+            value, factory.RelatedFactory
```

Comment:
We do another isinstqnce(..., RelatedFactory) in the else branch. Can you invert this if-else, and do this kind of check instead:

```python
if instance(..., (SubFactory, RelatedFactory)):
    ...
elif isinstance(..., PostGeneration):
    ...
elif isinstance(..., PostGenenerationMethodCall)
    ...
else:
raise NotImplementedError(useful error message to the user)
```